### PR TITLE
Add unit tests for sharing data store

### PR DESCRIPTION
**Pull request by manzoorwanijk**
Part of #34182

## Proposed changes:

* Add tests for the logic created in #34183

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Run `jetpack test` for `publicize-components`

### CODE REVIEW

File: projects/js-packages/publicize-components/src/hooks/use-share-limits/test/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+import { SOCIAL_STORE_CONFIG, SOCIAL_STORE_ID } from '../../../social-store';
+import { SocialStoreState } from '../../../social-store/types';
+import { createActiveConnections } from '../../../utils/test-utils';
+
+const messages = {
+	default: 'Share limit for 30 days: 30.',
+	limitExceeded: 'You have exceeded your share limit. Your posts will not longer be shared.',
+	scheduled: 'Your scheduled posts will not get shared after you reach the sharing limit.',
+	approachingLimit: 'You are approaching your share limit.',
+};
```
Comment (gmjuhasz) [inline projects/js-packages/publicize-components/src/hooks/use-share-limits/test/index.test.tsx]: Should export the messages in the main hook and import them here? That way it's more robust on changes on phrasing.

File: projects/js-packages/publicize-components/src/social-store/selectors/test/shares-data.test.js [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+				numberOfSharesRemaining( {
+					sharesData: {
+						is_share_limit_enabled: true,
+						publicized_count: 35,
+					},
+				} )
+			).toBe( 0 );
+		} );
+
+		const suites = [
```
Comment (gmjuhasz) [inline projects/js-packages/publicize-components/src/social-store/selectors/test/shares-data.test.js:199]: Maybe we could import big arrays like this from a `constants` or `utils` file? I think the tests look cleaner and easier to read like that, but I have no hard opinion on this, so feel free to leave it like this if you prefer keeping them together

File: projects/js-packages/publicize-components/src/hooks/use-share-limits/test/index.test.tsx (hunk 1 above)
Comment (manzoorwanijk) [reply projects/js-packages/publicize-components/src/hooks/use-share-limits/test/index.test.tsx]: Done. Thanks

File: projects/js-packages/publicize-components/src/social-store/selectors/test/shares-data.test.js (hunk 2 above)
Comment (manzoorwanijk) [reply projects/js-packages/publicize-components/src/social-store/selectors/test/shares-data.test.js:199]: I think the test data is better if it's inline. It provides more context about the test.

**Review by gmjuhasz — APPROVED** [review]
Looks good to me. Nice work!

**Comment by manzoorwanijk** [comment]
Since this PR was approved and there were no changes to it after that apart from the base branch update, I am going to merge it into the feature branch.

**Pull request merged**